packages/block-editor/src/components/block-lock/style.scss

```diff
@@ -22,7 +22,6 @@
 	.components-base-control__field {
 		align-items: center;
 		display: flex;
```

Comment:
Not in this PR, but a good follow-up for these would be to move the Icon out of the CheckboxControl `label` (it doesn't make sense), and move these flex styles into the parent element so we can remove these hacky overrides on `components-base-control__field`.

(Same with `block-manager`)

Reply:
@mirka when I initially read your comment, I assumed it was related to accessibility. I've just tested with my screen reader and looked a bit into it, but I couldn't find anything. Would you mind sharing why it is best to move the icon out of the label (aside from the opportunity to remove the CSS overrides)?

Reply:
Correct, it's not an a11y issue, just that there is no reason for the aria-hidden icon to be in the `label`. The main motivation is to remove the CSS override. I'll add some context in case some of it was missing:

Any consumer usage that reaches into and overrides the internal styling of a wp-component is inherently brittle. If BaseControl changes its structure or styling at some point, these BlockLock overrides could break in unexpected ways. We've been discouraging style hacks like this, since it actively harms our ability to make changes to wp-components without things breaking all over the place. It's doubly harmful because third-party devs often turn to the Gutenberg source code to learn how they should be building things.

So when things look like you might want to override internal wp-component styling, the decision flow would roughly be like:

1. Can I accomplish the same thing without adding a style hack? If so, that is always better.
2. Is this a common design pattern that will appear multiple times in the Gutenberg app? If so, consider proposing a change to the wp-component in question so it can flexibly handle more use cases out of the box.
3. If no to either of those questions, consider building your own component from scratch, or modifying your design so it works within the existing component limitations.

Reply:
Thank you for the detailed explanation! Very helpful! 🙌